feat: adds location and url to calendar link [#43]

roman-bytes opened the pull request:
Closes #43 

- Adds `location` and `url` to the calendar invite link
- Separated computed data and template
- Removes simple-icon package and just uses svgs

BenDMyers commented on src/schedule/schedule.11tydata.js:
```diff
@@ -1,45 +1,65 @@
+				start,
+				duration: [1, 'hour'],
+				location,
+				url,
```
Unfortunately, I'm not sure Google Calendar supports `url` — it might just be Outlook. What would you say about doing a bit of a switcheroo for Google Calendars, where we use the URL for the location?

```
location: url
```

chantastic replied:
interesting. do you think this is something that might need to be considered at the calendar-link library level?

i had assumed that they would work because we use the url field for all of our meetings (Google org).

roman-bytes replied:
The calendar-link library mentions the following statement on their page:

> The url field defaults to document.URL if a global document object exists. For server-side rendering, you should supply the url manually.

> Not all calendars support the guests and url fields.

I wonder if we could do a zoom type thing and add this info into the description?

chantastic commented:
Jacob! this is awesome! i'm so delighted to take that conversation we all had and transform it into working code!

i like the removal of the `simpleicons` dependency into static svgs a lot. i think that'll be a good pattern moving forward.

where would be a good place to document this decision? Readme? Wiki? Other?

BenDMyers commented:
Prepending the URL to the description is a great idea — guarantees everyone
gets it, and for people who get it twice, redundancy isn’t a bad thing.

roman-bytes commented:
@chantastic I am not sure what the best place to document this would be. Maybe in the wiki somewhere on how to use the SVGs / how to add a new SVG.

I was planning on making the `url` plus `description` change during the live-event.

BenDMyers commented:
> I was planning on making the `url` plus `description` change during the live-event.

In that case, is there anything more we need for this pull request, or can we merge this in? @roman-bytes @chantastic

chantastic commented:
hey @roman-bytes!

super sorry for the miscommunication here.
when i had suggested we do this feature as a live event, i was thinking it would be for the whole addition of the `calendar-link` functionality.

but we kinda got carried away in building 😅

i'd still love for you to do this as a stream tho.
do you have any interest in re-doing the feature as an educational stream?

we could create a branch, back out the feature, and you could re-build it with the advantage of everything we learned along the way.

if so, my suggestion would be this:
* let's add the url items in this PR
* create an event for a talk `Add Calendar Links to 11ty with Computed Data`
* create a branch with the feature removed and add the whole feature

Thoughts on proceeding that way?

roman-bytes commented:
@chantastic Once I realized how simple this was to do. I realized what a solid fit this would be for a lunch session. So I agree, I will get this PR updated with the description to include the URL and we can plan on doing an educational session with this feature.

roman-bytes commented:
Added things with latest commit
- Only show calendar links for future events and not past
- Passed a `isPastEvent` boolean along with calendar link data
- Added partial for calendar link section (renders based on boolean)
- If there is a description add in the url to the discord page

BenDMyers reviewed (approved):
This looks great, @roman-bytes! Thank you so much for putting this together for us! Please feel free to merge this in whenever you're ready!
